pluton/spawn/containercache/sshcache.go

```diff
@@ -0,0 +1,278 @@
+package containercache
+
+import (
+	"bytes"
+	"fmt"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/coreos/mantle/platform"
+	"github.com/coreos/mantle/util"
+)
+
+const (
+	cacheDir          = "/home/core/containercache"
+	downloadRetries   = 3
+	downloadRetryWait = 15 * time.Second
+	downloadTimeout   = 2 * time.Minute
+)
+
+var once sync.Once
+
+type ImageName struct {
+	Name   string
+	Engine string // docker or rkt
+}
+
+// StartBastionOnce will populate a bastion machine with the given containers
+// for rkt and docker. Since its best to run this only once per test suite, the
+// bastion machine should be in a separate platform.Cluster then the one used
+// in the test so it can be available for all tests in a given run. This
+// function is guarded with a sync.Once, so only the first within a test suite
+// it will have it run.
+func StartBastionOnce(bastion platform.Machine, names []ImageName) error {
+	var err error
+
+	f := func() {
+		err = startBastion(bastion, names)
+	}
+	once.Do(f)
+
+	return err
+}
+
+// Pull all containers cached by the bastion node into the docker and rkt store
+// of the given machines.
+func Pull(bastion platform.Machine, machines []platform.Machine) error {
```

Comment:
This seems misnamed since ultimately the copy is performed by a push from bastion to the target machine. Perhaps Load or similar that doesn't imply direction would be more appropriate.